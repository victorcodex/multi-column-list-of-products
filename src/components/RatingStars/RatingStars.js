import styled from "styled-components";

const RatingStars = styled.span`
  position: relative;
  display: inline-block;
  &:before {
    position: absolute;
    top: 0;
    left: 0;
    content: '★★★★★';
    color: #d3d3d3;
  }
  &:after {
    position: absolute;
    top: 0;
    left: 0;
    content: '★★★★★';
    color: #000000;
    overflow: hidden;
    width: ${props => props.rating}%
`;

export default RatingStars;