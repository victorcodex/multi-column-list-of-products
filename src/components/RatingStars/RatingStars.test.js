import React from 'react';
import { render } from '@testing-library/react';
import RatingStars from './RatingStars';

describe('RatingStars component', () => {
  it('should render rating stars', () => {
    const { asFragment } = render(<RatingStars rating={100}>★★★★★</RatingStars>);
    expect(asFragment()).toMatchSnapshot();
  });
});