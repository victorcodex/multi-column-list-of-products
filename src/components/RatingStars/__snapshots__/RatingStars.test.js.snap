// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingStars component should render rating stars 1`] = `
<DocumentFragment>
  <span
    class="sc-bdnxRM bFTQSW"
  >
    ★★★★★
  </span>
</DocumentFragment>
`;
